### PR TITLE
Support for File-Based Inputs for Live Streaming #38. Implemented MP4…

### PR DESCRIPTION
**Pull request by abhjaw**
…_FILE option for livestream

*Issue #, if available: #38 

*Description of changes:*
Implemented MP4_FILE option for livestream. This workflow now creates a MediaLive input with the mp4 file URL input parameter. The input is attached to a newly created MediaLive channel and corresponding MediaPackage destination. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: provider-utils/livestream-questions.json [hunk 1]
```diff
@@ -270,9 +274,19 @@
...
               "required": true
-          }
+            },
+            {
+                "key": "mp4URL",
+                "question": "Provide URL to the MP4 file: ",
+                "validation": {
+                    "operator": "regex",
+                    "value": "^[a-z0-9_:\\\/]+$",
+                    "onErrorMsg": "mp4URL can only be a valid url"
```
Comment (wizage) [inline provider-utils/livestream-questions.json]: You can make error message more user friendly. An example: `Error the provided URL is not valid`

File: provider-utils/awscloudformation/cloudformation-templates/livestream-helpers/LambdaFunctions/psdemo-js-live-workflow_v0.4.0/lib/flagfish.js [hunk 2]
```diff
@@ -236,13 +239,21 @@ class Flagfish extends mxStoreResponse(class { }) {
     if (this.ingestType === 'RTMP_PUSH') {
       payload.Destinations = ['p', 'b'].map(x => ({ StreamName: `${this.channelId}-${x}` }));
     }
+     /* Set MediaLive input Source as the Url to MP4 file*/
+    if (this.ingestType === 'MP4_FILE') {
+      payload.Sources =  [{ Url: this.mp4URL }, {Url: this.mp4URL}];
```
[the remark this answers is not in the document]
Comment (abhjaw) [reply provider-utils/awscloudformation/cloudformation-templates/livestream-helpers/LambdaFunctions/psdemo-js-live-workflow_v0.4.0/lib/flagfish.js]: I agree. I suggest shipping this out and add the support for backup URL at a later date. Hope you agree.

File: provider-utils/livestream-questions.json (hunk 1 above)
Comment (abhjaw) [reply provider-utils/livestream-questions.json]: Fixed the error message.

File: provider-utils/awscloudformation/cloudformation-templates/livestream-helpers/LambdaFunctions/psdemo-js-live-workflow_v0.4.0/lib/flagfish.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (wizage) [reply provider-utils/awscloudformation/cloudformation-templates/livestream-helpers/LambdaFunctions/psdemo-js-live-workflow_v0.4.0/lib/flagfish.js]: Sounds great! Merging!

**Comment by wizage** [comment]
Will push to live soon!

**Comment by abhjaw** [comment]
Thanks, Sam. That's awesome!

**Pull request merged**